src/train.jl

```diff
@@ -0,0 +1,23 @@
+function cb(opt_state:: Optim.OptimizationState)
+    cur_pred = collect(predict_adjoint(opt_state.metadata["x"]))
+    n = size(training_data, 1)
+    pl = scatter(1:n,training_data[:,10],label="data", legend =:bottomright,title="Spatial Plot at t=$(saveat[10])")
+    scatter!(pl,1:n,cur_pred[:,10],label="prediction")
+    pl2 = scatter(saveat,training_data[N÷2,:],label="data", legend =:bottomright, title="Timeseries Plot at Middle X")
+    scatter!(pl2,saveat,cur_pred[N÷2,:],label="prediction")
+    display(plot(pl, pl2, size=(600, 300)))
+    display(opt_state.value)
+    false
+end
+
+cb(trace::Optim.OptimizationTrace) = cb(last(trace))
+
+function sciml_train!(loss_f, params, data, opt::Optim.AbstractOptimizer;cb = cb)
```

Comment:
Does this actually work? Needs tests

Reply:
Can you suggest a test I should use?

Reply:
https://github.com/JuliaDiffEq/DiffEqFlux.jl#universal-differential-equations @ChrisRackauckas the example here with Flux and Optim should be a good MWE to build on?

Reply:
Yes exactly. It should be pretty clear what the interface is to most users of the package. I think the confusion might be just from not using it. Give the tutorials and everything a run, tweak them around, play with them, and see the use case is like. Then the training loop changes explanation will make sense, and you'll see what needs to be done with this PR. Right now this PR doesn't make any sense.